api/graphql/schema.py

```diff
@@ -169,6 +177,47 @@ def resolve_queryset(
         ).distinct()
 
 
+class ApplicationEventsFilter(AuthFilter, django_filters.FilterSet):
+    permission_classes = (
+        (ApplicationEventPermission,)
+        if not settings.TMP_PERMISSIONS_DISABLED
+        else [AllowAny]
+    )
+
+    @classmethod
+    def resolve_queryset(
+        cls, connection, iterable, info, args, filtering_args, filterset_class
+    ):
+        queryset = super().resolve_queryset(
+            connection, iterable, info, args, filtering_args, filterset_class
+        )
+
+        if settings.TMP_PERMISSIONS_DISABLED:
+            return queryset
+
+        # Filtering queries formation
+        user = info.context.user
+        unit_ids = user.unit_roles.filter(
+            role__permissions__permission="can_validate_applications"
+        ).values_list("unit", flat=True)
+        group_ids = user.unit_roles.filter(
+            role__permissions__permission="can_validate_applications"
+        ).values_list("unit_group", flat=True)
+        units = Unit.objects.filter(
+            Q(id__in=unit_ids) | Q(unit_groups__in=group_ids)
+        ).values_list("id", flat=True)
+
+        return queryset.filter(
+            Q(
```

Comment:
I have to somehow remember these cryptic one-letter functions like `Q`, `V` and `F` Looks useful but it probably takes a while until I remember to use them 😅

Reply:
The `Q` here is a class for making db queries and here it is used because want to have `OR` condition => so the `Q`s can be used in AND/OR operations and then can be passed to `.filter` like:
`condition = (Q(field_one__gte=else) & Q(field_one__lte=other) | Q(field_too=False)`
`Model.objects.filter(condition)`